src/model/consent/consent.ts

```diff
@@ -83,20 +83,44 @@ export class ConsentDB {
   }
 
   // Update Consent
-  // No validation against Null or illegal updates in models
+  // Only validating against updating existing fields
   public async update (consent: Consent): Promise<number> {
```

Comment:
maybe we should consider returning `boolean` value instead of `number` and be consistent with the way we do for `insert` method?

Reply:
@eoln we return the `updateCount` here is a useful check of how many rows have been changed. The reason `insert` returns a `boolean` is because MySQL and SQLite don't return anything useful on `insert`. So we just had to go with a `boolean`.

Reply:
hmm, but we are using limit in transaction context, so after an update, the update count should be always 1 - am I right?

Reply:
Yes, it should be 1 always because `Consent` is unique. But the `update` on MySQL and SQLite returns number of rows affected. This also ensures coherence with `Scope` where `update` can return any number of affected rows.

Reply:
don't get me wrong @ramanmangla but your code is inside transaction -> it means all atomic SQL commands are totally isolated. So if you do a select in L93 with and there is a uniqueness constraint on relation Consent-Scope you will always get one item, also you will always update one item in L120 ->the transaction guarantee that, so your code will always return 1, so maybe it is good to consider using boolean instead which will be the simpler concept.

Reply:
but maybe it is better to keep interface consistency with Scopes.update, so leave it as it is ;-)

Reply:
I agree with you. Yes, we can leave it for now for consistency purposes.